### PR TITLE
feat: refactor git discrad flow 

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Added the git rebase to the discard flow. This is to avoid the push failure when due to branch protection rules. 
Prev, it was discard and pull. This flow had above mentioned issue where if the branch protection rule was present, then the local branch would be in a bad state. 

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/23989

#### Type of change
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?
- [ ] Manual

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by nayan-rafiq — DISMISSED** [review]
LGTM

File: app/client/cypress/support/gitSync.js [hunk 1]
```diff
@@ -412,7 +412,7 @@ Cypress.Commands.add("gitDiscardChanges", () => {
   //cy.wait(6000);
   cy.get(gitSyncLocators.discardChanges)
     .children()
-    .should("have.text", "Discard changes");
```
Comment (AnaghHegde) [inline app/client/cypress/support/gitSync.js:415]: This is supposed to be Discard & Rebase right?

Comment (AnaghHegde) [reply app/client/cypress/support/gitSync.js:415]: @brayn003 This is supposed to be Discard & Rebase right?

**Review by nayan-rafiq — APPROVED** [review]

**Pull request merged**